manifests/params.pp

```diff
@@ -311,14 +317,23 @@
     default  => '/etc/default/puppetserver',
   }
 
-  $server_jvm_java_bin = '/usr/bin/java'
-  $server_jvm_min_heap_size = '2G'
-  $server_jvm_max_heap_size = '2G'
-  $server_jvm_extra_args = '' # lint:ignore:empty_string_assignment
+  $server_jvm_java_bin      = '/usr/bin/java'
+  $server_jvm_min_heap_size = '4G'
+  $server_jvm_max_heap_size = '4G'
+  $server_jvm_extra_args    = '' # lint:ignore:empty_string_assignment
 
-  $server_ssl_dir_manage = true
-  $server_default_manifest = false
-  $server_default_manifest_path = '/etc/puppet/manifests/default_manifest.pp'
+  $server_ssl_dir_manage           = true
+  $server_default_manifest         = false
+  $server_default_manifest_path    = '/etc/puppet/manifests/default_manifest.pp'
   $server_default_manifest_content = '' # lint:ignore:empty_string_assignment
+  $server_max_active_instances     = $::processorcount
+  $server_idle_timeout             = 1200000
+  $server_connect_timeout          = 120000
+  $server_enable_ruby_profiler     = false
+  $server_ca_auth_required         = true
+  $server_admin_api_whitelist      = [ '127.0.0.1', '::1', $::ipaddress ]
+  $server_ca_client_whitelist      = [ '127.0.0.1', '::1', $::ipaddress ]
+  $server_cipher_suites            = [ 'TLS_RSA_WITH_AES_256_CBC_SHA256', 'TLS_RSA_WITH_AES_256_CBC_SHA', 'TLS_RSA_WITH_AES_128_CBC_SHA256', 'TLS_RSA_WITH_AES_128_CBC_SHA', ]
+  $server_ssl_protocols            = [ 'TLSv1', 'TLSv1.1', 'TLSv1.2', ]
```

Comment:
I'd argue to only use TLSv1.2 because I'd expect all puppetserver implementations run on sufficiently new software versions, but I have no idea how realistic that expectation is.